src/components/Quote.js

```diff
@@ -0,0 +1,69 @@
+import React, { useState, useEffect } from 'react';
+
+const Quote = () => {
+  const [newQuote, setNewQuote] = useState(null);
+  const [quoteLoading, setQuoteLoading] = useState(true);
+  const [error, setError] = useState('');
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const API_KEY = 'aQHetjM7fPp6GpfmgJe4mA==zGAdLfljeQtfgrXo';
+      const quoteCategory = 'health';
+
+      try {
+        const res = await fetch(`https://api.api-ninjas.com/v1/quotes?category=${quoteCategory}`, {
+          headers: {
+            'X-Api-Key': API_KEY,
+          },
+        });
+
+        if (!res.ok) {
+          throw new Error('There is problem with the Network');
+        }
+
+        const resultData = await res.json();
+        const randIndex = Math.floor(Math.random() * resultData.length);
+        const randomQuote = resultData[randIndex];
+        setNewQuote(randomQuote);
+        setQuoteLoading(false);
+      } catch (error) {
+        setError(error.message);
+        setQuoteLoading(false);
+      }
+    };
+
+    if (!newQuote) {
+      fetchData();
+    }
+  }, [newQuote]);
```

Comment:
- [ ] Great work on getting the fetching part done! It looks like it might be fetching twice, though. This could be related to how the if statement and useEffect are interacting. Would you mind taking a closer look to debug it? Check out the video below to get a clearer idea of what might be going on:

   [scrnli_16_9_2023_11-54-26.webm](https://github.com/MohammadYaser/math-magicians/assets/107952724/cf0a742a-d76c-46c1-8a4b-cb43b8df1ad3)